main.go

```diff
@@ -201,9 +211,36 @@ func LoadBundles (wh warehouse.Warehouse, filename string, bundles ...fullstory.
 	return nil
 }
 
+func getExportData(fs *fullstory.Client, bundleID int) (fullstory.ExportData, error) {
+	log.Printf("Getting Export Data for bundle %d\n", bundleID)
+	for r := 1; r <= maxAttempts; r++ {
+		stream, err := fs.ExportData(bundleID)
+		if err == nil {
+			return stream, nil
+		}
+
+		log.Printf("Failed to fetch export data for Bundle %d", bundleID)
+		retryAfterDuration := defaultRetryAfterDuration
+		if statusError, ok := err.(fullstory.StatusError); ok {
+			// If the status code is NOT 429 and the code is below 500 we will not attempt to retry
+			if statusError.StatusCode != http.StatusTooManyRequests && statusError.StatusCode < 500 {
+				break
+			}
+
+			if retryAfter, err := strconv.Atoi(statusError.RetryAfter); err == nil {
+				retryAfterDuration = time.Duration(retryAfter) * time.Second
+			}
+		}
+
+		log.Printf("Attempt #%d failed. Retrying after %s\n", r, retryAfterDuration)
```

Comment:
My apologies, I should have noticed this before, but this function never logs or returns the original error that occurred in the call to `fs.ExportData`. Maybe this line is a good spot to log it? 

In addition to logging the error on each attempt, I think it would be good to return the last error that occurred, or to wrap it.  You can do this with https://godoc.org/github.com/pkg/errors#example-Wrap. It looks like we already have`github.com/pkg/errors` in our dependencies.

Reply:
good point. It would be very frustrating for people to not see client errors! done!